### PR TITLE
[BugFix] 1D query fix for MoE models

### PR DESCRIPTION
**Pull request by njhill**
MoE models were broken by #3236.

### CODE REVIEW

**Comment by njhill** [comment]
@rkooo567 FYI

I guess we should look into how this got through CI tests, maybe it would be covered by the ones currently failing due to the CUDA reinitialization issue.

File: vllm/model_executor/models/deepseek.py [hunk 1]
```diff
@@ -150,11 +150,11 @@ def pack_params(self):
         self.w2 = self.w2.view(len(w2), *w2s[0].shape)
 
     def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
-        batch_size, sequence_length, hidden_dim = hidden_states.shape
+        sequence_length, hidden_dim = hidden_states.shape
```
Comment (WoosukKwon) [inline vllm/model_executor/models/deepseek.py:153]: Could we rename `sequence_length` to `num_tokens`?

File: vllm/model_executor/models/mixtral.py [hunk 2]
```diff
@@ -124,9 +124,9 @@ def weight_loader(self, param: nn.Parameter, loaded_weight: torch.Tensor,
             param_data[expert_id, :, :] = loaded_weight[:, shard]
 
     def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
-        batch_size, sequence_length, hidden_size = hidden_states.shape
+        sequence_length, hidden_size = hidden_states.shape
```
Comment (WoosukKwon) [inline vllm/model_executor/models/mixtral.py:127]: ditto

File: vllm/model_executor/models/mixtral_quant.py [hunk 3]
```diff
@@ -132,9 +132,9 @@ def __init__(
                                      linear_method=None)
 
     def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
-        batch_size, sequence_length, hidden_dim = hidden_states.shape
+        sequence_length, hidden_dim = hidden_states.shape
```
Comment (WoosukKwon) [inline vllm/model_executor/models/mixtral_quant.py:135]: ditto

**Review by WoosukKwon — COMMENTED** [review]
@njhill Thanks for finding and fixing the bug! I think the MoE models were skipped from the tests because of their large sizes. 😓

**Comment by njhill** [comment]
@WoosukKwon updated per your suggestion, also fixed the moe test.

**Review by WoosukKwon — APPROVED** [review]
LGTM! Thanks again for the fix!

**Comment by rkooo567** [comment]
@njhill thanks for catching this, and sorry I was missing this from my PR!

**Comment by rkooo567** [comment]
Wonder if there's a better way to improve test coverages here. I also found the performance in general is not well tracked. 

In Ray, we have had nightly testing infra to report benchmark result & larger scale testing with lower frequency than every CI. Do you guys have any plan to do something similar?

**Comment by njhill** [comment]
@rkooo567 yes we have some with our non-vLLM based stack that we are transitioning off, and so will be using that with vLLM too.

**Pull request merged**